### ISSUE TITLE
request for evalution/inference code

### ISSUE THREAD

**Issue by Michael4933** [Visual-Agent/DeepEyes#16]
Thanks for your great work. Could you provide the direct inference code for evaluation? Integrating automate bbox output parsing and target region cropping, then come to the final answer.

**Comment by jiangsongtao** [comment]
+1. Thanks for your great work and wait  for the evalution code.

**Comment by JaaackHongggg** [comment]
Thanks for your interest on our work! 

We are continuing cleaning our code and we will release the evaluation code asap. 

Please stay tuned!

Best regards,
Jack

**Comment by JaaackHongggg** [comment]
Hi, we have update the evaluation code. You can refer to [here](https://github.com/Visual-Agent/DeepEyes/blob/main/eval/EVALUATION.md) for more details. 

Thanks again for your interest. If you have any question, please feel free to let we know.

Best regards,
Jack

**Comment by ROIM1998** [comment]
Thanks for the great work! Do you have any updates or plans to share the evaluation code for HR-Bench? Thanks a lot!

**Comment by JaaackHongggg** [comment]
Hi, we have updated the evaluation code for HR-Bench. See the Evaluation for more details.

**Comment by Alisaleelee** [comment]
> Hi, we have update the evaluation code. You can refer to [here](https://github.com/Visual-Agent/DeepEyes/blob/main/eval/EVALUATION.md) for more details.
> 
> Thanks again for your interest. If you have any question, please feel free to let we know.
> 
> Best regards, Jack

![Image](https://github.com/user-attachments/assets/97333e60-6b28-4696-a0a3-2736266ed942)
hello, when I used eval_vstrar.py, this error occurred. Why is that? Will it affect the final evaluation result?

**Comment by dddraxxx** [comment]
> > Hi, we have update the evaluation code. You can refer to [here](https://github.com/Visual-Agent/DeepEyes/blob/main/eval/EVALUATION.md) for more details.
> > Thanks again for your interest. If you have any question, please feel free to let we know.
> > Best regards, Jack
> 
> ![Image](https://github.com/user-attachments/assets/97333e60-6b28-4696-a0a3-2736266ed942) hello, when I used eval_vstrar.py, this error occurred. Why is that? Will it affect the final evaluation result?

This could be solved adding `--limit_mm_per_prompt "image=10"` when using vllm

**Comment by dddraxxx** [comment]
> Hi, we have update the evaluation code. You can refer to [here](https://github.com/Visual-Agent/DeepEyes/blob/main/eval/EVALUATION.md) for more details.
> 
> Thanks again for your interest. If you have any question, please feel free to let we know.
> 
> Best regards, Jack

What is the vllm command you use to serve your model ChenShawn/DeepEyes-7B, I use sth like 
```
vllm serve $trained_weights \
    --port 18900 \
    --gpu-memory-utilization 0.8 \
    --tensor-parallel-size 1 \
    --served-model-name "de7b" \
    --trust-remote-code \
    --disable-log-requests \
    --limit_mm_per_prompt "image=10"
``` 
but the model output is bad and contains weird tokens like `addCriterion`

**Comment by JaaackHongggg** [comment]
Sorry for late response. The problem of `At most 1 images may be provided in one request` can be solved add `--limit_mm_per_prompt "image=10"` . You can refer to https://github.com/Visual-Agent/DeepEyes/issues/16#issuecomment-2932584578 for details.

For the weird tokens problem, this may be resulted from the different version of vllm. The vllm we used is 0.8.2. You can try ot use the same version of vllm. Besides, we also met with the similar problem during developing. However, for our released ckpt, we did not meet with this problem.

If you have any question, please feel free to let we know.

Best regards,
Jack

**Comment by JaaackHongggg** [comment]
@hongge831 Your problem looks like the similar to https://github.com/Visual-Agent/DeepEyes/issues/16#issuecomment-2933780773. You can have a try.

**Comment by hongge831** [comment]
@jiangsongtao thanks for soon reply, when using the "stop": ["<|im_end|>\n".strip(), "</tool_call>"] will indeed indeed truncate the output, but isn’t this just treating the symptoms and not the root cause?

**Comment by lky-violet** [comment]
Hello,

I have a question about using the following `stop` parameter:

```python
"stop": ["<|im_end|>\n".strip(), "</tool_call>"]
```
When using the Transformers library to load and run a model, where exactly should this parameter be added in the code? #19 

Should it be passed to the generate() function or set elsewhere?

Thank you!

**Comment by pspdada** [comment]
> > [@hongge831](https://github.com/hongge831) You can have a try. We think the reason of the chaotic output is primarily due to engineering issues. For example, precision loss may occur during the parameter saving and conversion process of verl, or it may be caused by versions of vllm. It is also possible that the chaotic output is due to Qwen 2.5 VL not being trained on similar data during the pre-training phase.
> 
> [@JaaackHongggg](https://github.com/JaaackHongggg) Hi! Thanks for your awesome contribution! My vlllm version is `0.8.2`, and I met the same problem as the repeated output and weird output, i.e., 'addCriterion'. And i try as [#16 (comment)](https://github.com/Visual-Agent/DeepEyes/issues/16#issuecomment-2933780773). It indeed solve the repeated output. But it lead to another problem.
> 
> ```
> {"role": "assistant", "content": "<think>The umbrella is on the right side of the traffic light. The traffic light is located to the left of the umbrella, which is positioned further to the right in the image.</think>  \n<tool_call>\n{\"name\": \"image_zoom_in_tool\", \"arguments\": {\"bbox_2d\": [1732, 1163, 1930, 1238], \"label\": \"umbrella\"}}\n"}, {"role": "user", "content": [{"type": "image_url", "image_url": {"url": "data:image/jpeg;base64,"}}, {"type": "text", "text": "\nThink first, call **image_zoom_in_tool** if needed, then answer. Format strictly as:  <think>...</think>  <tool_call>...</tool_call> (if tools needed)  <answer>...</answer> "}]}, {"role": "assistant", "content": "<think>The umbrella is indeed on the right side of the traffic light, as shown in the zoomed-in image.</think>  \n<answer>A. The umbrella is on the right side of the traffic light.</answer>  \n<tool_call>\n{\"name\": \"image_zoom_in_tool\", \"arguments\": {\"bbox_2d\": [1732, 1163, 1930, 1238], \"label\": \"umbrella\"}}\n"}, {"role": "user", "content": [{"type": "image_url", "image_url": {"url": "data:image/jpeg;base64,"}}, {"type": "text", "text": "\nThink first, call **image_zoom_in_tool** if needed, then answer. Format strictly as:  <think>...</think>  <tool_call>...</tool_call> (if tools needed)  <answer>...</answer> "}]}
> ```
> 
> The model invoke a second tool-call, although the answer is already got, which is not expected. Do you have any idea? By the way, the acc results is
> 
> ```
> Accuracy for direct_attributes: 89.57%
> Accuracy for relative_position: 85.53%
> Overall Accuracy: 87.96%
> ```
> 
> which is a little different from the reported results.

The result I tested using the official checkpoint is:

```text
Accuracy for direct_attributes: 84.21%
Accuracy for relative_position: 84.21%
Overall Accuracy: 84.21%
error_nums_type: {'direct_attributes': 0, 'relative_position': 0}
```

It seems a bit weird.

**Comment by Michael4933** [comment]
> Hello,
> 
> I have a question about using the following `stop` parameter:
> 
> "stop": ["<|im_end|>\n".strip(), "</tool_call>"]
> When using the Transformers library to load and run a model, where exactly should this parameter be added in the code? [#19](https://github.com/Visual-Agent/DeepEyes/issues/19)
> 
> Should it be passed to the generate() function or set elsewhere?
> 
> Thank you!

a vey simple yet not elegant method:
directly modify `_sample` function in `transformers.generation.utils.py`：

            unfinished_sequences = unfinished_sequences & ~stopping_criteria(input_ids, scores)
            this_peer_finished = unfinished_sequences.max() == 0
            if next_tokens == torch.tensor([[151645]]).cuda() or next_tokens == torch.tensor([[151658]]).cuda():
                this_peer_finished =True
specific token_id could be obtained by processor.tokenizer.encode('</tool_call>'>) or '<|im_end|>

Good luck~
